tests/unit/hook_packages/terraform/test_prepare_hook.py

```diff
@@ -2572,3 +2599,98 @@ def test_get_python_command_name_python_not_found(self, mock_run_side_effect, mo
         expected_error_msg = "Python not found. Please ensure that python 3.7 or above is installed."
         with self.assertRaises(PrepareHookException, msg=expected_error_msg):
             _get_python_command_name()
+
+    def test_generate_makefile_rule_for_lambda_resource(self):
+        sam_metadata_resource = SamMetadataResource(
+            current_module_address=None, resource={"address": "null_resource.sam_metadata_aws_lambda_function"}
+        )
+        makefile_rule = _generate_makefile_rule_for_lambda_resource(
+            python_command_name="python",
+            output_dir="/some/dir/path/.aws-sam/output",
+            sam_metadata_resource=sam_metadata_resource,
+            terraform_application_dir="/some/dir/path",
+            logical_id="function_logical_id",
+        )
+        expected_makefile_rule_regex = (
+            r"^build-function_logical_id:(\n|\r\n)"
+            r"\tterraform apply -target null_resource\.sam_metadata_aws_lambda_function -auto-approve(\n|\r\n)"
+            r"\tterraform show -json \| python .+ "
+            r'\|planned_values\|root_module\|resources\[\?address=="null_resource\.sam_metadata_aws_lambda_function"\]'
+            r"\|values\|triggers\|built_output_path \$\(ARTIFACTS_DIR\) \/some\/dir\/path(\n|\r\n)"
+        )
+        self.assertRegex(makefile_rule, expected_makefile_rule_regex)
+
+    def test_build_show_command_rule(self):
+        sam_metadata_resource = SamMetadataResource(current_module_address=None, resource={})
+        show_command = _build_show_command_rule(
+            python_command_name="python",
+            output_dir="/some/dir/path/.aws-sam/output",
+            resource_address="null_resource.sam_metadata_aws_lambda_function",
+            sam_metadata_resource=sam_metadata_resource,
+            terraform_application_dir="/some/dir/path",
+        )
+        script_path = Path(".aws-sam", "output", "copy_terraform_built_artifacts.py")
+        expected_show_command = (
+            f"terraform show -json | python {script_path} "
+            '|planned_values|root_module|resources[?address=="null_resource.sam_metadata_aws_lambda_function"]'
+            "|values|triggers|built_output_path $(ARTIFACTS_DIR) /some/dir/path"
+        )
+        self.assertEqual(show_command, expected_show_command)
```

Comment:
It seems like these two tests actually include testing of functionality of the child functions that they call, rather than just testing the functionality of that function itself. For example, if you changed the output of `build_jpath_string`, you'd have to update `test_build_jpath_string`, `test_build_show_command_rule` and `test_generate_makefile_rule_for_lambda_resource` (rather than just `test_build_jpath_string`).

You can work around this by mocking function calls and just asserting they're called, rather than testing for their output.